app/controllers/users_controller.rb

```diff
@@ -22,10 +30,54 @@ def show
     redirect_to action: :new
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user.update_attributes user_params
+      flash[:success] = t "success.update_profile"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    user = User.find_by id: params[:id]
+    if user
+      user.destroy
+      flash[:success] = t "success.delete_user"
+    else
+      flash[:danger] = t "error.invalid_ID"
+    end
+    redirect_to users_url
+  end
+
   private
 
   def user_params
     params.require(:user)
           .permit :name, :email, :password, :password_confirmation
   end
+
+  def logged_in_user
+    return if logged_in?
+
+    store_location
+    flash[:danger] = t "error.pls_login"
+    redirect_to login_path
+  end
+
+  def correct_user
+    @user = User.find_by id: params[:id]
+    return if current_user? @user
+
+    redirect_to root_url
+  end
+
+  def admin_user
+    redirect_to root_url unless current_user.admin?
```

Comment:
```suggestion
    redirect_to root_url unless current_user.admin
```
ở đây gọi thuộc tính admin của đối tượng user mới đúng chứ em? thấy khai báo tên thuộc tính là admin mà

Reply:
Dạ, nó là kiểu boolean nên thêm dấu ? cũng ko có vấn đề gì mà chị, em nghĩ tường minh hơn chút

Reply:
`admin?` đây là 1 giá trị hay là 1 method em tự định nghĩa đây? 
nếu là giá trị của field thì ko có `?` còn method em định nghĩa là thì có `?` là ok đúng chuẩn nhé :p